### PR TITLE
Implement rfc #1437

### PR DESCRIPTION
**Pull request by ticki**
https://github.com/rust-lang/rfcs/pull/1437

### CODE REVIEW

File: src/libcore/char.rs [hunk 1]
```diff
@@ -306,6 +306,7 @@ impl CharExt for char {
     fn escape_default(self) -> EscapeDefault {
         let init_state = match self {
             '\t' => EscapeDefaultState::Backslash('t'),
+            '\x1B' => EscapeDefaultState::Backslash('e'),
```
Comment (tomjakubowski) [inline src/libcore/char.rs]: this literal could be changed to `'\e` after a snapshot, correct? might be worth noting

Comment (tomjakubowski) [inline src/libcore/char.rs]: Can this can be changed to `'\e'` after a snapshot? Might be worth noting.

File: src/libcoretest/char.rs [hunk 2]
```diff
@@ -34,6 +34,7 @@ fn test_is_whitespace() {
     assert!('\n'.is_whitespace());
     assert!(!'a'.is_whitespace());
     assert!(!'_'.is_whitespace());
+    assert!(!'\e'.is_whitespace());
```
Comment (tomjakubowski) [inline src/libcoretest/char.rs]: How does this build w/ a snapshot compiler?

File: src/libcore/char.rs (hunk 1 above)
Comment (Ticki) [inline src/libcore/char.rs]: Yeah. When bootstrapping, we can substitute it to `\e`.

**Comment by alexcrichton** [comment]
Thanks for the PR! For now I'm going to close this in favor of the RFC, but this is linked up so it should be easy to find if we end up accepting.

**Pull request closed without merging**